cibuildwheel/windows.py

```diff
@@ -455,7 +455,9 @@ def build(options: Options, tmp_path: Path) -> None:
                             tmp_file.write_bytes(constraints_path.read_bytes())
                             constraints_path = tmp_file
 
-                        build_env["PIP_CONSTRAINT"] = str(constraints_path)
+                        build_env["PIP_CONSTRAINT"] = (
+                            str(constraints_path) + " " + build_env.get("PIP_CONSTRAINT", "")
+                        )
```

Comment:
```suggestion
                        constraint = build_env.get("PIP_CONSTRAINT", "")
                        build_env["PIP_CONSTRAINT"] = f"{constraints_path} {constraint}"
```

Reply:
```suggestion
                        constraints = [str(constraints_path), build_env.get("PIP_CONSTRAINT", None)]
                        build_env["PIP_CONSTRAINT"] = " ".join(filter(constraints, 
```

Reply:
Yeah, this is a bit unclear. I'm feeling something like this:
```python
                        our_constraints = str(constraints_path)
                        user_constraints = build_env.get("PIP_CONSTRAINT")
                        build_env["PIP_CONSTRAINT"] = " ".join(
                            c for c in [our_constraints, user_constraints] if c
                        )
```

Reply:
Sure, that's basically what I wrote below. `filter(item, None)` filters None's out, by the way, or you can the c for c if c pattern.

Reply:
I think it's `filter(None, a_list)`, but yeah, never been used map or filter in python though, list comprehensions are more readable and powerful imo!